ticketbot.py

```diff
@@ -73,85 +71,73 @@ def get_links(match_set, sha_validation=validate_sha_github):
     return links
 
 
-class TicketBot(irc.IRCClient):
-    """A bot for URLifying Django tickets."""
-
-    nickname = "ticketbot"
-    password = os.environ['NICKSERV_PASS']
-    channels = os.environ['CHANNELS'].split(',')
-
-    def connectionMade(self):
-        irc.IRCClient.connectionMade(self)
-
-    def connectionLost(self, reason):
-        irc.IRCClient.connectionLost(self, reason)
+@irc3.plugin
+class Plugin:
 
-    def signedOn(self):
-        """Called when bot has succesfully signed on to server."""
-        self.setNick(self.nickname)
-        self.msg('NickServ', 'identify %s' % (self.password))
-        for channel in self.channels:
-            self.join(channel)
+    def __init__(self, bot):
+        self.bot = bot
 
-    def privmsg(self, user, channel, msg):
-        """This will get called when the bot receives a message."""
-        user = user.split('!', 1)[0]
-        matches = get_matches(msg)
+    @irc3.event(irc3.rfc.PRIVMSG)
+    def process_msg_or_privmsg(self, mask, event, target, data, **kw):
+        """Detect special markers, reply with their respective links."""
+        # We shouldn't send automatic replies to notices
```

Comment:
Avoid use of "we" in comments per our [style guide](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style), e.g. “Don't send..."